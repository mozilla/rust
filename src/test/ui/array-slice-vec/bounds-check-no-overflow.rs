// run-fail
// error-pattern:index out of bounds
// ignore-emscripten no processes

#![warn(unconditional_panic)]

use std::mem::size_of;

fn main() {
    let xs = [1, 2, 3];
    xs[usize::MAX / size_of::<isize>() + 1]; //~ WARN: this operation will panic at runtime
}
