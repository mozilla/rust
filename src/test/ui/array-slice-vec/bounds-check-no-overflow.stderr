warning: this operation will panic at runtime
  --> $DIR/bounds-check-no-overflow.rs:11:5
   |
LL |     xs[usize::MAX / size_of::<isize>() + 1];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ index out of bounds: the length is 3 but the index is 2305843009213693952
   |
note: the lint level is defined here
  --> $DIR/bounds-check-no-overflow.rs:5:9
   |
LL | #![warn(unconditional_panic)]
   |         ^^^^^^^^^^^^^^^^^^^

warning: 1 warning emitted

